sqlmesh/core/context.py

```diff
@@ -2148,6 +2157,36 @@ def _load_requirements(self, path: Path) -> None:
                         )
                     self._requirements[dep] = ver
 
+    def _load_multiple_gateways(self) -> None:
+        # Validate there are multiple gateways and the default is duckdb
+        if (
+            len(self.config.gateways) < 2
+            or not isinstance(self._connection_config, DuckDBConnectionConfig)
+            or not self._connection_config.database
+            or self._connection_config.catalogs
+        ):
+            return
```

Comment:
Should we actually validate more strictly here? If the cross-gateway conditions aren't met, then it seems like we should just fail and emit a warning, otherwise the error handling is deferred to post-load time. Unless I'm missing something?

Reply:
I thought in the rare case where the cross_gateway is set to true but the user only has one gateway, or maybe a different type than the supported ones, not to fail, but simply continue with the default gateway as normal. But I'm happy to make the validation stricter and emitting an error when the conditions aren't met